shared/reducers/chat.js

```diff
@@ -13,25 +13,39 @@ const initialConversation: ConversationState = new ConversationStateRecord()
 // dedupes and removed deleted messages. Applies edits
 function _processMessages (seenMessages: Set<any>, messages: List<ServerMessage> = List(), prepend: List<ServerMessage> = List(), append: List<ServerMessage> = List(), deletedIDs: Set<any>): {nextSeenMessages: Set<any>, nextMessages: List<ServerMessage>} {
   const filteredPrepend = prepend.filter(m => !seenMessages.has(m.key))
-  const filteredAppend = append.filter(m => !seenMessages.has(m.key) && m.type !== 'Edit')
+  const filteredAppendGroups = append.filter(m => !seenMessages.has(m.key)).groupBy(m => m.type === 'Edit' || m.type === 'UpdateAttachment' ? m.type : 'Append')
+  const filteredAppend = filteredAppendGroups.get('Append') || List()
 
   const messagesToUpdate = Map(prepend.concat(append).filter(m => seenMessages.has(m.key)).map(m => [m.key, m]))
   const updatedMessages = messages.map(m => messagesToUpdate.has(m.key) ? messagesToUpdate.get(m.key) : m)
   // We have to check for m.messageID being falsey and set.has(undefined) is true!. We shouldn't ever have a zero messageID
   let nextMessages: List<ServerMessage> = filteredPrepend.concat(updatedMessages, filteredAppend).filter(m => !m.messageID || !deletedIDs.has(m.messageID))
   const nextSeenMessages = Set(nextMessages.map(m => m.key))
 
-  // Process edits
-  // $ForceType
-  const edits: List<EditingMessage> = append.filter(m => !seenMessages.has(m.key) && m.type === 'Edit')
-  edits.forEach(edit => {
-    const entry = nextMessages.findEntry(m => m.messageID === edit.targetMessageID)
+  filteredAppendGroups.get('Edit', List()).forEach(edit => {
+    if (edit.type !== 'Edit') {
```

Comment:
When would this not be edit.type === 'Edit', or is this just to make flow happy?